pkgs/build-support/node/build-npm-package/npm.patch

```diff
@@ -0,0 +1,83 @@
+diff --git a/lib/node_modules/npm/node_modules/@npmcli/run-script/lib/set-path.js b/lib/node_modules/npm/node_modules/@npmcli/run-script/lib/set-path.js
+index c59c270d9..34f59a623 100644
+--- a/lib/node_modules/npm/node_modules/@npmcli/run-script/lib/set-path.js
++++ b/lib/node_modules/npm/node_modules/@npmcli/run-script/lib/set-path.js
+@@ -12,7 +12,7 @@ const setPATH = (projectPath, binPaths, env) => {
+     .reduce((set, p) => set.concat(p.filter(concatted => !set.includes(concatted))), [])
+     .join(delimiter)
+ 
+-  const pathArr = []
++  const pathArr = [nodeGypPath, PATH]
```

Comment:
Maybe add a comment explaining that this ensures Node's built-in node-gyp is at the front of the `PATH` (to prevent packages overriding it with their own node-gyp dep, presumably)

The comment can be front matter at the top of the patch file

Reply:
I added this inline like the others -- any reason you think it shouldn't be?

Reply:
Inline is fine with me, as long as it's somewhere :)